File: la/6565/6565.py
N = 1001;
f=[0,1]
for i in range(2,N):
    f += [2 * f[i-2] + 3]
    j = 0
    while j <= i - 1 and 2**j < f[i]:
        f[i] = min(f[i], 2*(2**j - 1 + f[i-1-j]) +1)
        j += 1
kase=0
try:
    while True:
        n = int(input())
        kase += 1
        print("Case ",kase,": ",f[n],sep='')
except EOFError:
    pass

File: la/6565/6565.cpp
#include<bits/stdc++.h>
using namespace std;
typedef long long ll;
const int N = 1001;
ll f[N];
int main()
{
    f[0] = 0;
    f[1] = 1;
    for (int i = 2; i < N; ++i) {
        f[i] = 2 * f[i-1] + 1;
		for (int j = 1; j <= i - 1 && (1LL<<j) < f[i]; ++j)
			f[i] = min(f[i], 2 * ((1LL<<j) - 1 + f[i-1-j]) + 1);
	}
    int n,kase=0;
    while (scanf("%d",&n) == 1)
        printf("Case %d: %lld\n",++kase,f[n]);
}
